pkgs/by-name/gi/gitlab-ci-local/package.nix

```diff
@@ -26,6 +26,33 @@ buildNpmPackage rec {
       --replace-fail "npm run cleanup" "true"
   '';
 
+  postInstall = ''
+    NODE_MODULES=$out/lib/node_modules/gitlab-ci-local/node_modules
+
+    # Remove intermediate build files for re2 to reduce dependencies.
+    #
+    # This does not affect the behavior. On npm `re2` does not ship
+    # the build directory and downloads a prebuilt version of the
+    # `re2.node` binary. This method produces the same result.
+    cp $NODE_MODULES/re2/build/Release/re2.node re2.node
+    strip -x re2.node
+    rm -rf $NODE_MODULES/re2/build
+    install -Dt $NODE_MODULES/re2/build/Release re2.node
```

Comment:
Can we use rm with shopt to exlcude all files except re2.node?

Reply:
Thank you for the review!
I was not aware of using shopt to exclude files.
However, I do not think shopt makes it much cleaner.

Here is an example using shopt. Since the glob only works for the current directory, rm needs to be called multiple times to clean the directory.
```shell
(
  shopt -s extglob
  cd $NODE_MODULES/re2/build/Release
  rm -rf !(re2.node)
  cd ..
  rm -rf !(Release)
)
strip $NODE_MODULES/re2/build/Release/re2.node
```

There is another way using find which looks better but leaves some empty directories.
```shell
find $NODE_MODULES/re2/build -type f ! -name re2.node -delete
strip $NODE_MODULES/re2/build/Release/re2.node
```

Let me know if you prefer one option or if I missed a better one.

Reply:
Done.
I had to use the `-path` option because two files had identical names.